### PR TITLE
Bug 2116572: OSDOCS-3268 Auto-include instance types

### PR DESCRIPTION
**Pull request by codyhoag**
https://issues.redhat.com/browse/OSDOCS-3268

This is a POC for the work I've collaborated on with Engineering and QE leadership. Google doc for a bit of context:

https://docs.google.com/document/d/1XGZQRJshZD2V4eqLLflDflzGuYIn6zSzXIV9kHKoGIs/edit

### CODE REVIEW

**Comment by yunjiang29** [comment]
The instance types' content: https://github.com/openshift/installer/pull/5517

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.11` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.10. And any PR going into `main` must also target the latest version branch (enterprise-4.11).

If the update in your PR does NOT apply to version 4.11 onward, please retarget this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by yunjiang29** [comment]
please @MayXuQQ (Azure) @jianli-wei (GCP) @aleskandro (ARM) help review, thanks.

File: modules/installation-aws-arm-tested-machine-types.adoc [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// aws modules
+
+[id="installation-aws-arm-tested-machine-types_{context}"]
+= Tested instance types for AWS ARM
+
+The following Amazon Web Services (AWS) ARM instance types have been *tested* with {product-title}.
```
Comment (yunjiang29) [inline modules/installation-aws-arm-tested-machine-types.adoc:8]: Do we need to emphasize the word *tested* using a bold style?

File: modules/installation-aws-arm-tested-machine-types.adoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// aws modules
+
+[id="installation-aws-arm-tested-machine-types_{context}"]
+= Tested instance types for AWS ARM
```
Comment (yunjiang29) [inline modules/installation-aws-arm-tested-machine-types.adoc:6]: the original title is `Supported AWS machine types`, now we changed it to `Test ...` , my concern is that, users might ask "why isn't the supported list?"

@cuppett @patrickdillon PTAL, thanks.

**Comment by MayXuQQ** [comment]
> @yunjiang29, can we merge this change?

modules/installation-azure-tested-machine-types.adoc need change, now in https://github.com/openshift/installer/pull/5704, list the vm family of the instance type which I tested, not the machine type.

**Comment by MayXuQQ** [comment]
@kalexand-rh  https://github.com/openshift/installer/pull/5704 merged, list the vm family , not the machine type. need update https://deploy-preview-41796--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-customizations.html#installation-azure-tested-machine-types_installing-azure-customizations

File: modules/installation-gcp-tested-machine-types.adoc [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Module included in the following assemblies:
+//
+// gcp modules
+
+[id="installation-gcp-tested-machine-types_{context}"]
+= Tested instance types for GCP
+
+The following Microsoft Azure instance types have been *tested* with {product-title}.
```
Comment (jianli-wei) [inline modules/installation-gcp-tested-machine-types.adoc:8]: GCP instead?

**Review by jianli-wei — CHANGES_REQUESTED** [review]

**Comment by sadasu** [comment]
/retitle Bug 2116572: OSDOCS-3268 Auto-include instance types

File: modules/installation-gcp-tested-machine-types.adoc (hunk 3 above)
Comment (MayXuQQ) [reply modules/installation-gcp-tested-machine-types.adoc:8]: now list the instance family ? @jianli-wei

Comment (MayXuQQ) [reply modules/installation-gcp-tested-machine-types.adoc:8]: @sadasu  could you help update it ?

File: modules/installation-aws-arm-tested-machine-types.adoc [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+The following Amazon Web Services (AWS) ARM instance types have been *tested* with {product-title}.
+
+.Machine types based on arm64 architecture
+[%collapsible]
+====
+include::https://raw.githubusercontent.com/openshift/installer/master/docs/user/aws/tested_instance_types_aarch64.md[]
+====
```
Comment (MayXuQQ) [inline modules/installation-aws-arm-tested-machine-types.adoc:14]: @yunjiang29  the tested_instance_types_aarch64.md is family name, shall we change the md file name?

**Comment by bergerhoffer** [comment]
The enterprise-4.12 label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.11. And any PR going into `main` must also target the latest version branch (enterprise-4.12).

If the update in your PR does NOT apply to version 4.12 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by kalexand-rh** [comment]
Closing in favor of https://github.com/openshift/openshift-docs/pull/49799

**Pull request closed without merging**